### PR TITLE
[Brent] [Waste] Add upcoming renewal email opt-in control for garden sub and renew.

### PR DESCRIPTION
**Pull request by neprune**
[skip changelog]

closes https://github.com/mysociety/societyworks/issues/4202

# Preview

renewal control

renewal summary

### CODE REVIEW

File: perllib/FixMyStreet/Cobrand/Kingston.pm [hunk 1]
```diff
@@ -47,6 +47,8 @@ sub waste_payment_ref_council_code { "RBK" }
 
 sub garden_collection_time { '6:30am' }
 
+sub garden_subscription_email_renew_reminder_opt_in { 0 }
```
Comment (dracos) [inline perllib/FixMyStreet/Cobrand/Kingston.pm:50]: Don't need this line as it inherits from CobrandSLWP

File: perllib/FixMyStreet/App/Controller/Waste.pm [hunk 2]
```diff
@@ -1477,6 +1477,11 @@ sub process_garden_renew : Private {
     $c->forward('setup_garden_sub_params', [ $data, $type ]);
     $c->forward('add_report', [ $data, 1 ]) or return;
 
+    if ($data->{email_renewal_reminders}) {
```
Comment (dracos) [inline perllib/FixMyStreet/App/Controller/Waste.pm:1480]: Could this (and 1538 similarly) be in `setup_garden_sub_params` so only needs to be in the one place? I know it's also called for cancellation etc but this check would still be there.

**Review by dracos — CHANGES_REQUESTED** [review]
Looks good, only couple of tiny suggestions :)

File: perllib/FixMyStreet/App/Controller/Waste.pm (hunk 2 above)
Comment (neprune) [reply perllib/FixMyStreet/App/Controller/Waste.pm:1480]: I think `setup_garden_sub_params` is called before we have the report we can attach the extra field to. Not sure there's any good alternative.

**Review by dracos — APPROVED** [review]
Thanks :) Will need to update the Brent open311-adapter Echo config on the server for 1159 (garden) to add the extra data field on staging.

**Pull request merged**